qiskit_optimization/problems/quadratic_program.py

```diff
@@ -998,6 +994,9 @@ def substitute_variables(
                 - Same variable is substituted multiple times.
                 - Coefficient of variable substitution is zero.
         """
+        # pylint: disable=cyclic-import
+        from .substitute_variables import SubstituteVariables
+
         return SubstituteVariables().substitute_variables(self, constants, variables)
```

Comment:
Given the usage here I guess it begs the question why a class rather than a function. The only reason I can see is that it avoids passing src/dst/subs to routines that would need it rather they are stored in the class. I am not suggesting to change things now, since it all works and there are other things todo. More just noting this since I noticed this while reviewing things.

Reply:
I will make it a function